qiskit/pulse/commands/_functional_pulse.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=missing-param-doc
+
+"""
+Functional pulse.
+"""
+
+import warnings
+from inspect import signature
+
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.pulse.commands._sample_pulse import SamplePulse
+
+
+class FunctionalPulse:
+    """Decorator of pulse envelope function."""
+
+    def __init__(self, pulse):
+        """Create new pulse envelope function.
+
+        Args:
+            pulse (callable): a function describing pulse envelope.
+            pulse function must contain argument "duration".
+
+        Raises:
+            QiskitError: when incorrect envelope function is specified
+        """
+
+        if callable(pulse):
+            sig = signature(pulse)
+            if 'duration' in sig.parameters:
+                self.pulse = pulse
+            else:
+                raise QiskitError('Pulse function requires "duration" argument.')
+        else:
+            raise QiskitError('Pulse function is not callable.')
+
+    def __call__(self, duration, **kwargs):
+        """Create new functional pulse.
+        """
+        return FunctionalPulseComm(self.pulse, duration=duration, **kwargs)
+
+
+class FunctionalPulseComm(SamplePulse):
```

Comment:
Let's not abbreviate class names
```suggestion
class FunctionalPulseCommand(SamplePulse):
```